spec/client_spec.rb

```diff
@@ -0,0 +1,379 @@
+#
+# Author:: Chef Partner Engineering (<partnereng@chef.io>)
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'spec_helper'
+
+describe Vra::Client do
+  describe '#request_headers' do
+    context 'when bearer token exists' do
+      it 'has an Authorization header' do
+        @vra.bearer_token = '12345'
```

Comment:
a common rspec patter would be to use `described_classz and `subject ` for tests like this, eg:

```ruby
describe Vra::Client do
  subject { described_class.new }

  describe '#request_headers' do
    context 'when bearer token exists' do
      it 'has an Authorization header' do
        subject.bearer_token = '12345'
        expect(subject.request_headers.key?('Authorization')).to be true
      end
    end
  end
end